lib/OpenQA/WebAPI/Plugin/AMQP.pm

```diff
@@ -0,0 +1,181 @@
+# Copyright (C) 2016 SUSE Linux LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::WebAPI::Plugin::AMQP;
+
+use strict;
+use warnings;
+
+use parent qw/Mojolicious::Plugin/;
+use JSON;
+use Mojo::IOLoop;
+use OpenQA::Utils;
+use OpenQA::Schema::Result::Jobs;
+use Mojo::RabbitMQ::Client;
+
+my $app;
+my $config;
+my $client;
+my $channel;
+my $reconnecting = 0;
```

Comment:
Really not liking these globals. As far as i can see they could all just be attributes of the instance of this class (`$self->{client}`, `$self->{reconnecting}` and so on).